lib/internal/options.js

```diff
@@ -43,7 +47,7 @@ function refreshOptions() {
 
 function getOptionValue(optionName) {
   const options = getCLIOptionsFromBinding();
-  if (optionName.startsWith('--no-')) {
+  if (StringPrototypeStartsWith(optionName, '--no-')) {
```

Comment:
Maybe better avoiding it by doing 

```suggestion
  if (
    optionName.length > 5 &&
    optionName[0] === '-' &&
    optionName[1] === '-' &&
    optionName[2] === 'n' &&
    optionName[3] === 'o' &&
    optionName[4] === '-' 
  ) {
```